src/common/sizer.cpp

```diff
@@ -676,15 +677,34 @@ wxSizer::~wxSizer()
 
 wxSizerItem* wxSizer::DoInsert( size_t index, wxSizerItem *item )
 {
-    m_children.Insert( index, item );
+    // Use the scoped pointer to avoid memory leaking when
+    // wxWindowBase::SetContainingSizer() throws.
+    wxScopedPtr<wxSizerItem> scoped_item( item );
 
     if ( item->GetWindow() )
-        item->GetWindow()->SetContainingSizer( this );
+    {
+#if wxUSE_EXCEPTIONS
+        try
+        {
+#endif
+            item->GetWindow()->SetContainingSizer( this );
+#if wxUSE_EXCEPTIONS
+        }
+        catch ( ... )
+        {
+            item->DetachWindow();
```

Comment:
Sorry, I don't understand why is this necessary. Without this call, and try/catch around it, if `SetContainingSizer()` throws, the scoped pointer containing `item` will delete it, and call `SetContainingSizer(NULL)` on the window -- but what's wrong with that?